### PR TITLE
[RTM] The Contao users must only be loaded if the correct scope is active

### PR DESCRIPTION
**Pull request by aschempp**
fixes https://github.com/contao/standard-edition/issues/7

### CODE REVIEW

**Comment by leofeyer** [comment]
Don't forget to add `[RTM]` if you are finished.

**Comment by aschempp** [comment]
Should we add unit tests for that case?

**Comment by leofeyer** [comment]
Yes.

**Comment by leofeyer** [comment]
And please don't forget to fix the tests :)

**Comment by aschempp** [comment]
I have rebased the PR and fixed/improved the unit tests.

File: src/Security/User/ContaoUserProvider.php [hunk 1]
```diff
@@ -57,4 +59,20 @@ public function supportsClass($class)
     {
         return is_subclass_of($class, 'Contao\\User');
     }
+
+
+    private function isFrontend($username)
```
Comment (leofeyer) [inline src/Security/User/ContaoUserProvider.php]: I'd prefer `isFrontendUser()` and `isBackendUser()`.

**Comment by leofeyer** [comment]
Except for my one comment, the PR looks RTM to me :)

**Comment by aschempp** [comment]
Name updated and Scrutinizer issue fixed.

**Comment by leofeyer** [comment]
I could actually merge this PR if the tests weren't failing :)

**Comment by aschempp** [comment]
Checking…

File: src/Security/ContaoAuthenticator.php [hunk 2]
```diff
@@ -78,12 +63,16 @@ public function authenticateToken(TokenInterface $token, UserProviderInterface $
 
         try {
             $providerKey = $token->getKey();
-            $user        = $this->userProvider->loadUserByUsername($providerKey);
+            $user        = $userProvider->loadUserByUsername($providerKey);
```
Comment (leofeyer) [inline src/Security/ContaoAuthenticator.php]: IMHO, the code should be as follows:

```php
try {
    $user = $userProvider->loadUserByUsername($token->getKey());
} catch (UsernameNotFoundException $e) {
    return $token;
}

if ($user instanceof User) {
    return new ContaoToken($user);
}

return $token;
```

Comment (leofeyer) [inline src/Security/ContaoAuthenticator.php]: @aschempp Can I change the code as mentioned above?

Comment (aschempp) [inline src/Security/ContaoAuthenticator.php]: good idea, I just changed it.

**Comment by leofeyer** [comment]
Merged in eb6ecce875875aec1d1d584d7b67c09bdf3796d5.

**Comment by leofeyer** [comment]
@aschempp After merging this, I can open neither the back end nor the front end:

```
A Token was not found in the SecurityContext.
500 Internal Server Error - AuthenticationCredentialsNotFoundException
```

**Pull request merged**